src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesSynchronisationService.kt

```diff
@@ -20,35 +21,74 @@ class IncentivesSynchronisationService(
 
   suspend fun synchroniseIncentive(iepEvent: IncentiveUpsertedOffenderEvent) {
     nomisApiService.getIncentive(iepEvent.bookingId, iepEvent.iepSeq).run {
+
       // todo integration and remove this log
       log.debug("received nomis incentive: $this")
 
-      mappingService.findNomisIncentiveMapping(iepEvent.bookingId, iepEvent.iepSeq)?.run {
+      mappingService.findNomisIncentiveMapping(
+        nomisBookingId = bookingId,
+        nomisIncentiveSequence = incentiveSequence
+      )?.let { incentiveMapping ->
         // todo remove this log
-        log.debug("found nomis incentive mapping: $this")
+        log.debug("found nomis incentive mapping: $incentiveMapping")
+        incentiveService.synchroniseUpdateIncentive(
+          iepEvent.bookingId, incentiveMapping.incentiveId,
+          UpdateIncentiveIEP(
+            reviewTime = this.iepDateTime,
+            commentText = this.commentText,
+            current = this.currentIep
+          )
+        )
         telemetryClient.trackEvent(
           "incentive-updated-synchronisation",
           mapOf(
             "bookingId" to iepEvent.bookingId.toString(),
             "incentiveSequence" to iepEvent.iepSeq.toString(),
-            "auditModuleName" to iepEvent.auditModuleName
+            "auditModuleName" to iepEvent.auditModuleName,
+            "currentIep" to this.currentIep.toString()
           ),
           null
         )
+        if (!this.currentIep) {
```

Comment:
might be worth adding a comment on why we do this, I will forget by next week

Reply:
yes - will do